pytket/extensions/qiskit/backends/aer.py

```diff
@@ -373,228 +421,140 @@ def __init__(
             # architecture is non-trivial
-            passlist.append(
-                CXMappingPass(
-                    arch,
-                    NoiseAwarePlacement(
-                        arch,
-                        self._backend_info.averaged_node_gate_errors,
-                        self._backend_info.averaged_edge_gate_errors,
-                        self._backend_info.averaged_readout_errors,
-                    ),
-                    directed_cx=True,
-                    delay_measures=False,
-                )
+            self._required_predicates.append(
+                ConnectivityPredicate(characterisation.architecture)
             )
-            passlist.append(NaivePlacementPass(arch))
-            if optimisation_level == 0:
-                passlist.append(self.rebase_pass())
-            elif optimisation_level == 1:
-                passlist.append(SynthesiseTket())
-            else:
-                passlist.extend([CliffordSimp(False), SynthesiseTket()])
-        return SequencePass(passlist)
 
-    def process_circuits(
-        self,
-        circuits: Sequence[Circuit],
-        n_shots: Union[None, int, Sequence[Optional[int]]] = None,
-        valid_check: bool = True,
-        **kwargs: KwargTypes,
-    ) -> List[ResultHandle]:
-        """
-        See :py:meth:`pytket.backends.Backend.process_circuits`.
-        Supported kwargs: `seed`.
-        """
-        # discard result but useful to validate n_shots
-        Backend._get_n_shots_as_list(
-            n_shots,
-            len(circuits),
-            optional=False,
-        )
-        return super().process_circuits(circuits, n_shots, valid_check, **kwargs)
 
-    def get_pauli_expectation_value(
-        self,
-        state_circuit: Circuit,
-        pauli: QubitPauliString,
-        valid_check: bool = True,
-    ) -> complex:
-        """Calculates the expectation value of the given circuit using the built-in Aer
-        snapshot functionality.
-        Requires a simple circuit with default register qubits, and no noise model.
-
-        :param state_circuit: Circuit that generates the desired state
-            :math:`\\left|\\psi\\right>`.
-        :type state_circuit: Circuit
-        :param pauli: Pauli operator
-        :type pauli: QubitPauliString
-        :param valid_check: Explicitly check that the circuit satisfies all required
-            predicates to run on the backend. Defaults to True
-        :type valid_check: bool, optional
-        :return: :math:`\\left<\\psi | P | \\psi \\right>`
-        :rtype: complex
-        """
-        if self._noise_model:
-            raise RuntimeError(
-                (
-                    "Snapshot based expectation value not supported with noise model. "
-                    "Use shots."
-                )
-            )
+class AerStateBackend(_AerBaseBackend):
+    _persistent_handles = False
+    _supports_shots = True
+    _supports_counts = True
```

Comment:
Hi @trvto , was there a reason to make `AerStateBackend` and `AerUnitaryBackend` report support for shots and counts here?

Reply:
OK good to know. Apparently this caused some InQuanto tests to fail so we should revert it if possible (I made #89 ).

Reply:
I opened a PR to this effect

Reply:
Yeah we could do a 0.37.1 release.